### PR TITLE
Better unauth handling @sefk

### PR DESCRIPTION
**Pull request by sefk**
This fixes issue #727. Currently logging out in the admin pages dumps you into an error page with nowhere to go. Now, redirect home. It's not perfect but I'm not sure what better place to send them.

Please review @dhcole or @ultrasaurus.

### CODE REVIEW

**Comment by dhcole** [comment]
With #749, you can remove fc68bddca60b766c0f4085212807dae7434495f1

**Comment by sefk** [comment]
More ambitious fix, handling errors globally now with ```$(document).ajaxError()```. The 397c473
 commit message:

    handle ajax errors globally with auth special case

    Part of fix for #727. Per discussion with @dhcole, instead of handling
    auth errors only in the admin section, this general handler helps
    for the whole app. Now whenever we get an auth error, fire up login.

    Add handling in the nav to prevent the login modal from opening multiple
    times.

    Instead of displaying a hardcoded error message, display the error we
    get back in the entity to the user. So those messages should be worded a
    bit better now: "You must login..." instead of "Access denied."

@dhcole: I still need your help with some event handling. The problem is with the dashboard admin page. Steps to repro:

1. log in as an admin
2. navigate to the admin pages, select the dashboard
3. open up chrome dev tools
4. select logout. You should see two 403 errors
5. the login dialog comes up, good. log in as a **non-admin user**
6. Now you can't log in as anyone anymore

Basically by logging in as a non-admin user on an admin page a *second time* you're causing yet another 403 error. The pending event to bring up the login dialog causes a problem.

I tried hacks like using ```_.defer()``` or ```_.debounce()``` but those weren't reliable, and wouldn't be robust anyway.

File: assets/js/backbone/apps/login/controllers/login_controller.js [hunk 1]
```diff
@@ -59,11 +58,17 @@ Login = BaseController.extend({
...
         window.cache.userEvents.trigger("user:login:success", user);
-        self.cleanup();
       }).modal('hide');
+
+    });
+
+    // clean up no matter how the modal is closed
+    $('#login').bind('hidden.bs.modal', function () {
+      window.cache.userEvents.trigger("user:login:close");
+      self.cleanup();
```
Comment (sefk) [inline assets/js/backbone/apps/login/controllers/login_controller.js]: originally these actions were only happening on successful login, i.e. not when the modal was dismissed by the "X". Factoring out into it's own listener fixed that.

**Comment by dhcole** [comment]
:+1: looking good @sefk! I'll look into the multiple error issue this afternoon.

**Comment by dhcole** [comment]
@sefk I ran through those steps this morning, but I'm not sure I'm seeing the same problem. Here's what I did. Note there is a js error for `user` not defined in the `$('.activity-block .activity-feed')` template.

1. Log in as admin
    screen shot 2015-05-04 at 10 43 15 am
2. Go to the admin dashboard
    screen shot 2015-05-04 at 10 43 36 am
3. Log out
    screen shot 2015-05-04 at 10 43 53 am
4. Log in as nonadmin (Get the log in dialog again, but logged in)
   screen shot 2015-05-04 at 10 43 59 am
5. (Click profile) Still logged in
screen shot 2015-05-04 at 10 44 13 am

**Comment by sefk** [comment]
Now that I've merged off of current master which now includes @dhcole's fix (#766) this works fine for me in my dev env. I think it's ready.

**Comment by dhcole** [comment]
Nice work @sefk! Tested, everything seems in order. Thanks for tackling this. :ship:
